src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -128,6 +128,26 @@ public function exists($files)
     }
 
     /**
+     * Checks if files are real files.
+     *
+     * @param string|array|\Traversable $files A filename, an array of files, or a \Traversable instance to check
+     *
+     * @return bool true if the file is a real file, false otherwise
+     */
+    public function isFile($files)
+    {
+        foreach ($this->toIterator($files) as $file) {
```

Comment:
`exists` for an iterator considers that all of them must exists. This can be useful. I fail to see a use case for passing a list of paths and then knowing whether there are only files in it. Most of the time, you will need to decide for each of them separately IMO (and in some other cases, you can filter directories out when building your iterator, when you control it...)